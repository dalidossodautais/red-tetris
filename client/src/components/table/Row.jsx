import styled from "styled-components";

export default styled.div.attrs((props) => {
  const added = {};
  added.borderColor = props.theme.primaryTextColor;
  if (props.selected)
    added.backgroundColor = props.theme.primaryBackgroundColor;
  return added;
})`
  border-bottom: 1px solid ${(props) => props.borderColor};
  background-color: ${(props) => props.backgroundColor};
  display: flex;
  flex-direction: row;
  width: ${(props) => props.width};
`;
