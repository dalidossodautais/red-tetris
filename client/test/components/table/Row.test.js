import React from "react";
import renderer from "react-test-renderer";
import Row from "../../../src/components/table/Row";

describe("Row", () => {
  test("Standard", () => {
    const component = renderer.create(<Row />);
    // rendering
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
    // re-rendering
    tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
  test("Standard selected", () => {
    const component = renderer.create(<Row selected />);
    // rendering
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
    // re-rendering
    tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
