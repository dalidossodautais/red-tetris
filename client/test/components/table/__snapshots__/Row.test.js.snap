// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Row Standard 1`] = `
<div
  className="sc-bdfBwQ fWUSiw"
/>
`;

exports[`Row Standard 2`] = `
<div
  className="sc-bdfBwQ fWUSiw"
/>
`;

exports[`Row Standard selected 1`] = `
<div
  className="sc-bdfBwQ fWUSiw"
  selected={true}
/>
`;

exports[`Row Standard selected 2`] = `
<div
  className="sc-bdfBwQ fWUSiw"
  selected={true}
/>
`;
